dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k.cc

```diff
@@ -237,31 +173,47 @@ DecodeResult NvJpeg2000DecoderInstance::DecodeImplTask(int thread_idx,
   if (!ParseJpeg2000Info(in, ctx))
     return result;
 
-  const int64_t channels = ctx.shape[0];
+  const int64_t channels = ctx.image_info.num_components;
+  bool is_roi_oob = ctx.roi &&
+    (ctx.roi.begin[0] < 0 || ctx.roi.end[0] > ctx.image_info.image_width ||
+     ctx.roi.begin[1] < 0 || ctx.roi.end[1] > ctx.image_info.image_height);
+
   DALIImageType format = channels == 1 ? DALI_GRAY : DALI_RGB;
   bool is_processing_needed =
     channels > 1 ||  // nvJPEG2000 decodes into planar layout
     ctx.pixel_type != opts.dtype ||
     format != opts.format ||
-    (ctx.bpp != 8 && ctx.bpp != 16);
+    (ctx.bpp != 8 && ctx.bpp != 16) ||
+    is_roi_oob;
 
   auto decode_out = out;
+  auto decode_sh = ctx.orig_shape;
```

Comment:
nitpick, the assigned value is never used - it gets overwritten in both branches of the following if.